aeron-driver/src/main/c/media/aeron_receive_destination.c

```diff
@@ -41,14 +41,17 @@ int aeron_receive_destination_create(
     _destination->transport.data_paths = _destination->data_paths;
     _destination->local_sockaddr_indicator.counter_id = AERON_NULL_COUNTER_ID;
 
+    const size_t socket_sndbuf = 0 != channel->socket_sndbuf ? channel->socket_sndbuf : context->socket_sndbuf;
+    const size_t socket_rcvbuf = 0 != channel->socket_rcvbuf ? channel->socket_rcvbuf : context->socket_rcvbuf;
+
     if (context->udp_channel_transport_bindings->init_func(
         &_destination->transport,
         &channel->remote_data,
         &channel->local_data,
         channel->interface_index,
         0 != channel->multicast_ttl ? channel->multicast_ttl : context->multicast_ttl,
-        context->socket_rcvbuf,
-        context->socket_sndbuf,
+        socket_rcvbuf,
```

Comment:
Maybe we should pass the channel to be more efficient and future proof.

Reply:
Probably is a good clean up.  I think I will come back to it later as the current PR hasn't required a change there yet.  I am changing the congestion control supplier in a similar way that to pass through the `aeron_udp_channel_t` instead of the string value to avoid re-parsing.